sdk/core/azure-core/src/main/java/com/azure/core/util/CoreUtils.java

```diff
@@ -249,4 +251,32 @@ public static String bomAwareToString(byte[] bytes, String contentType) {
             }
         }
     }
+
+    /**
+     * Prioritize {@code applicationId} set in ClientOptions over {@link HttpLogOptions}.
+     *
+     * @param logOptions provided {@link HttpLogOptions}.
+     * @param clientOptions provided {@link ClientOptions}.
+     * {@code applicationId} is different in {@link HttpLogOptions} and {@link ClientOptions}.
+     * @return applicationId based on rule explained above.
+     *
+     */
+    public static String getApplicationId(HttpLogOptions logOptions, ClientOptions clientOptions) {
+
+        String applicationId;
+        String logApplicationId = null;
```

Comment:
You can use ternary operators to make these more readable. I was trying to understand the assignments.